packages/hydrogen/src/pagination/Pagination.ts

```diff
@@ -116,6 +117,14 @@ export function Pagination<NodesType>({
     }
   }, [transition.state]);
 
+  useEffect(() => {
+    // Clean up location state on initial load
+    navigate(window.location.toString(), {
+      replace: true,
+      state: {nodes: undefined, pageInfo: undefined},
+    });
+  }, []);
```

Comment:
Additionally, I don't think it will work properly if we fix the URL navigation, because it will erase the navigation state even for soft transitions. This means the paginated state will disappear when going back/forth, breaking scroll position.

In my opinion the issue mentioned by Helen is not that big of a deal. It _only_ happens if you hard refresh on a page, then have click _next_ or _previous_ on one of the elements. And even then, it's not bad data getting loaded, just data from both paginated components loading at the same time. Jarring yes, but not rendering bad data at all.